lib/recursive-methods.rb

```diff
@@ -36,10 +46,18 @@ def search(array, value)
     raise NotImplementedError, "Method not implemented"
 end
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
 def is_palindrome(s)
```

Comment:
👍 This works, but you have similar time/space issues with the above methods due to creating new arrays.